static/app/components/modals/sentryAppDetailsModal.spec.tsx

```diff
@@ -0,0 +1,170 @@
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(mockRequests.interaction).toHaveBeenCalledWith(
+      `/sentry-apps/${sentryApp.slug}/interaction/`,
+      expect.objectContaining({
+        method: 'POST',
+        data: {
+          tsdbField: 'sentry_app_viewed',
+        },
+      })
+    );
+  });
+
+  it('displays the Integrations description', function () {
+    renderMockRequests({sentryAppSlug: sentryApp.slug});
+
+    render(
+      <SentryAppDetailsModal
+        closeModal={jest.fn()}
+        isInstalled={false}
+        onInstall={jest.fn()}
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(screen.getByText(sentryApp.overview)).toBeInTheDocument();
+  });
+
+  it('closes when Cancel is clicked', function () {
+    renderMockRequests({sentryAppSlug: sentryApp.slug});
+
+    const handleCloseModal = jest.fn();
+
+    render(
+      <SentryAppDetailsModal
+        closeModal={handleCloseModal}
+        isInstalled={false}
+        onInstall={jest.fn()}
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    userEvent.click(screen.getByText('Cancel'));
+
+    expect(handleCloseModal).toHaveBeenCalled();
+  });
+
+  it('installs the Integration when Install is clicked', function () {
+    renderMockRequests({sentryAppSlug: sentryApp.slug});
+
+    const handleOnInstall = jest.fn();
+
+    render(
+      <SentryAppDetailsModal
+        closeModal={jest.fn()}
+        isInstalled={false}
+        onInstall={handleOnInstall}
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    userEvent.click(screen.getByRole('button', {name: 'Accept & Install'}));
+
+    expect(handleOnInstall).toHaveBeenCalled();
+  });
+
+  it('when the User does not have permission to install Integrations, does not display the Install button', function () {
+    renderMockRequests({sentryAppSlug: sentryApp.slug});
+
+    render(
+      <SentryAppDetailsModal
+        closeModal={jest.fn()}
+        isInstalled={false}
+        onInstall={jest.fn()}
+        organization={TestStubs.Organization({access: []})}
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(
+      screen.queryByRole('button', {name: 'Accept & Install'})
+    ).not.toBeInTheDocument();
+  });
+
+  it('when the Integration is installed, render the Install button disabled', function () {
+    renderMockRequests({sentryAppSlug: sentryApp.slug});
+
+    render(
+      <SentryAppDetailsModal
+        closeModal={jest.fn()}
+        isInstalled
+        onInstall={jest.fn()}
+        organization={TestStubs.Organization()}
+        sentryApp={sentryApp}
+      />
+    );
+
+    expect(screen.getByRole('button', {name: 'Accept & Install'})).toBeDisabled();
+  });
+
+  it('when the Integration requires no permissions, does not render permissions', function () {
```

Comment:
I think the sentence needs reversed here so the “it” prefix works.

right now it’s “it when …”